lib/android/app/src/main/java/com/reactnativenavigation/anim/NavigationAnimator.java

```diff
@@ -68,9 +90,81 @@ public void onAnimationEnd(Animator animation) {
                 }
             }
         });
-        set.playTogether(translationY, alpha);
-        set.setDuration(DURATION);
-        set.setInterpolator(ACCELERATE_INTERPOLATOR);
         set.start();
     }
+
+    @NonNull
+    private AnimatorSet getDefaultPopAnimation(View view) {
+        AnimatorSet set = new AnimatorSet();
+
+        ObjectAnimator alpha = ObjectAnimator.ofFloat(view, View.ALPHA, 1, 0);
+        alpha.setInterpolator(ACCELERATE_INTERPOLATOR);
+
+        ObjectAnimator translationY = ObjectAnimator.ofFloat(view, View.TRANSLATION_Y, 0, this.translationY);
+        translationY.setInterpolator(ACCELERATE_INTERPOLATOR);
+        translationY.setDuration(DURATION);
+        alpha.setDuration(DURATION);
+        set.playTogether(translationY, alpha);
+        return set;
+    }
+
+    public void setOptions(AnimationsOptions options) {
+        this.options = options;
+    }
+
+    private AnimatorSet createAnimation(AnimationOptions options, View view) {
```

Comment:
Seems like there's a lot of duplication here.
Could have been cleaner if `AnimationsOptions` exposed a `toList()` function which returned all animations as list, then we would be able to iterate over them and create the appropriate animation with some factory method.
This is totally optional :)